ping.cpp

```diff
@@ -331,20 +335,24 @@ bool ping_start(IPAddress adr, int count=0, int interval=0, int size=0, int time
           ((((float)transmitted - (float)received) / (float)transmitted) * 100.0)
     );
 
-    if (received) {
-        ping_resp pingresp;
-        log_i("round-trip min/avg/max/stddev = %.3f/%.3f/%.3f/%.3f ms\r\n", min_time, mean_time, max_time, sqrt(var_time / received));
-        pingresp.total_count = 10;
-        pingresp.timeout_count = 10;
-        pingresp.total_bytes = 1;
-        pingresp.total_time = mean_time;
-        pingresp.ping_err = 0;
-        return true;
-        //	ping_o->sent_function(ping_o, (uint8*)&pingresp);
-    }
-    return false;
+    ping_resp pingresp;
+    log_i("round-trip min/avg/max/stddev = %.3f/%.3f/%.3f/%.3f ms\r\n", min_time, mean_time, max_time, sqrt(var_time / received));
+    pingresp.total_count = count; //Number of pings
+    pingresp.resp_time = mean_time; //Average time for the pings
+    //pingresp.seqno = 0; //not relevant
+    pingresp.timeout_count = transmitted - received; //number of pings which failed
+    pingresp.bytes = size; //number of bytes received for 1 ping
+    pingresp.total_bytes = size * count; //number of bytes for all pings
+    pingresp.total_time = (millis() - ping_started_time) / 1000.0; //Time consumed for all pings; it takes into account also timeout pings
```

Comment:
can you explain a little the used of millis here?

Reply:
I found no clear documentation regarding what total_time is.
So taking into account the we have total_count and resp_time members, I understood/assumed that total_time should represent the cumulated time spent for any of the following:
- valid pings (which had a pong response)
- timeout pings (those pings which had no response)
- the small time spent between the pings

That's why I've stored the millis() value before the "while" loop, and after all pings were processed, I compute the total time in seconds.

total_time is probably not so relevant, but I considered as a low impact to compute it, than to mark it as zero or similar.

Reply:
https://github.com/iputils/iputils we should align with iputils if possible.
From what I see the time is measured in milliseconds and it's the total time since the ping command started until it ended..almost like operation overall status

Reply:
> the time is measured in milliseconds and it's the total time since the ping command started until it ended..almost like operation overall status


This is what I intended, but without knowing if this should really be. 
Right now in master "total_time" is mean_time (so average time of all successful pings) and it's not used at all, while "resp_time" is read & used (but not set with any value). Since we have all the other statistics information like "total_count" , "timeout_count", "resp_time", I thought that the the "total_time" could return the time consumed by the overall ping operation. 

I took a short look to: https://github.com/iputils/iputils , the ping code from there is huge :-s

On this branch, the most important is to fix missing "resp_time" . Some other statistics are used by branch #15 ( Issue #9 ) while total_time has no intended use for now...